File: src/browser/create-profile.js
#!/usr/bin/env node

import { chromium } from 'playwright';
import { join, dirname } from 'path';
import { fileURLToPath } from 'url';
import { mkdirSync } from 'fs';

const __dirname = dirname(fileURLToPath(import.meta.url));
const ROOT = join(__dirname, '..', '..');
const PROFILES_DIR = join(ROOT, 'profiles');

const name = process.argv[2];

if (!name) {
  console.error('Usage: npm run profile:create <name>');
  console.error('Example: npm run profile:create amazon');
  process.exit(1);
}

const profilePath = join(PROFILES_DIR, name);
mkdirSync(profilePath, { recursive: true });

console.log(`Opening browser with profile "${name}" at ${profilePath}`);
console.log('Log in to the site, then close the browser window to save the profile.\n');

const context = await chromium.launchPersistentContext(profilePath, {
  headless: false,
  viewport: { width: 1280, height: 720 },
  args: ['--disable-gpu'],
});

const page = context.pages()[0] || await context.newPage();
await page.goto('about:blank');

// Wait for the user to close the browser
await new Promise((resolve) => {
  context.on('close', resolve);
});

console.log(`\nProfile "${name}" saved to ${profilePath}`);

File: src/browser/create_profile.py
#!/usr/bin/env python3
"""Create a persistent browser login profile."""

import asyncio
import sys
from pathlib import Path

from playwright.async_api import async_playwright

ROOT = Path(__file__).parent.parent.parent
PROFILES_DIR = ROOT / "profiles"


async def main():
    if len(sys.argv) < 2:
        print("Usage: python -m src.browser.create_profile <name>", file=sys.stderr)
        print("Example: python -m src.browser.create_profile amazon", file=sys.stderr)
        sys.exit(1)

    name = sys.argv[1]
    profile_path = PROFILES_DIR / name
    profile_path.mkdir(parents=True, exist_ok=True)

    print(f'Opening browser with profile "{name}" at {profile_path}')
    print("Log in to the site, then close the browser window to save the profile.\n")

    pw = await async_playwright().start()
    context = await pw.chromium.launch_persistent_context(
        str(profile_path),
        headless=False,
        viewport={"width": 1280, "height": 720},
        args=["--disable-gpu"],
    )

    page = context.pages[0] if context.pages else await context.new_page()
    await page.goto("about:blank")

    # Wait for the user to close the browser
    closed = asyncio.get_event_loop().create_future()
    context.on("close", lambda: closed.set_result(True))
    await closed

    await pw.stop()
    print(f'\nProfile "{name}" saved to {profile_path}')


if __name__ == "__main__":
    asyncio.run(main())
